digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -513,6 +513,34 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     }
   }
 
+  def createRoadLinkProject(roadAddressProject: RoadAddressProject) = {
+    withDynTransaction {
+      val id = Sequences.nextViitePrimaryKeySeqValue
+      var project = roadAddressProject.copy(id= id)
+
+      try {
+        val roadAddressExists = RoadAddressDAO.getRoadAddressProjectById(project.id) match{
+          case None => {
+            RoadAddressDAO.createRoadAddressProject(project)
+            //create ProjectLink
+            if(project.startPart <= project.endPart){
+              var part = project.startPart
+              while(part <= project.endPart) {
```

Comment:
you could remove var part and replace while row with: 
for (part<-project.startPart) {